src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -341,18 +352,32 @@ internal bool IsHostHealthy(bool throwWhenUnhealthy = false)
             return true;
         }
 
-        private bool ShutdownHostIfUnhealthy()
+        private bool ShutdownHostIfUnhealthy(ILogger logger)
         {
             if (ShouldMonitorHostHealth && _healthCheckWindow.GetEvents().Where(isHealthy => !isHealthy).Count() > _healthMonitorOptions.Value.HealthCheckThreshold)
             {
-                // if the number of times the host has been unhealthy in
-                // the current time window exceeds the threshold, recover by
-                // initiating shutdown
-                var message = $"Host unhealthy count exceeds the threshold of {_healthMonitorOptions.Value.HealthCheckThreshold} for time window {_healthMonitorOptions.Value.HealthCheckWindow}. Initiating shutdown.";
-                _logger.LogError(0, message);
+                logger = logger ?? _logger;
                 var environment = _rootServiceProvider.GetService<IScriptJobHostEnvironment>();
-                environment.Shutdown();
```

Comment:
This environment was resolving to null before the service registration fix above